tests/test_codemoduleapi.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+from nitsm.codemoduleapi import code_module, SemiconductorModuleContext
+
+
+@pytest.mark.pin_map("empty.pinmap")
+class TestCodeModuleApi:
+    """
+    Pytest passes fixtures as keyword arguments while the code_module decorator expects the TSM
+    context to be passed as a positional argument. Therefore, we must redirect the keyword argument
+    to a positional argument by creating surrogate methods.
+    """
+
+    @staticmethod
+    @code_module
+    def _static_method(tsm_context):
```

Comment:
The one case I don't see tested here is when you have arguments after the tsm_context. I realize that we test that case in other unit tests, but I think it makes sense to include it here with the code_module test cases.

Reply:
524e4e669760d5d1c19e2475d959b974d5b6f7e0

Reply:
Also added several more tests in the latest commit.